detailed_docs/TRS/RC/Module Allocation rules.md

```diff
@@ -0,0 +1,78 @@
+### Module Allocation rules
+
+To allocate a module mobile app sends one of the following RPCs
+* SetInteriorVehicleData 
+* ButtonPress
+
+1.
+In case 
+
+RC app sends RPC(moduleType + moduleId) request to allocate a free module  
+and `userLocation` is within `moduleInfo.serviceArea`
+
+SDL must  
+allocate the module and send a successfull response 
+
+2.
+
+In case 
+
+RC app sends RPC(moduleType + moduleId) request to allocate a free module  
+and `userLocation` is NOT within `moduleInfo.serviceArea`
+
+SDL must  
+reject allocation module request
+
+3.
+In case 
+
+RC app sends RPC(moduleType + moduleId) request to allocate a module already allocated by another app  
+and `moduleInfo.allowMultipleAccess=false`
+
+SDL must  
+reject allocation module request
+
+4.
+In case 
+
+RC app sends RPC(moduleType + moduleId) request to allocate a module already allocated by another app  
+and `moduleInfo.allowMultipleAccess=true`  
+and `userLocation` is NOT within `moduleInfo.serviceArea`
+
+SDL must  
+reject allocation module request
+
+5.
+In case 
+
+RC app sends RPC(moduleType + moduleId) request to allocate a module already allocated by another app  
+and `moduleInfo.allowMultipleAccess=true`  
+and `userLocation` is within `moduleInfo.serviceArea`  
+and RCAccessMode = AUTO_ALLOW
+
+SDL must  
+allocates the module(`moduleType` + `moduleId`) without asking the driver
+
+6.
+In case 
+
+RC app sends RPC(`moduleType` + `moduleId`) request to allocate a module already allocated by another app  
+and `moduleInfo.allowMultipleAccess=false`  
+and `userLocation` is within `moduleInfo.serviceArea`  
+and RCAccessMode = AUTO_DENY
+
+SDL must
+reject allocation module request with resultCode:IN_USE
```

Comment:
@KhrystynaDubovyk I think we need to add  `without asking the driver` like the above, for pointing attention to that.